helga_jeopardy.py

```diff
@@ -298,13 +556,109 @@ def jeopardy(client, channel, nick, message, cmd, args,
         reset_channel(channel, mongo_db)
         return 'done'
 
+
     # if we have an active question, and args, evaluate the answer
 
     question = mongo_db.find_one({
         'channel': channel,
         'active': True,
     })
 
+    new_game = mongo_db.find_one({
+        'channel': channel,
+        'game_active': True,
+        'game_started': False,
+    })
+
+    current_game = mongo_db.find_one({
+        'channel': channel,
+        'game_active': True,
+        'game_started': True,
+    })   
+
+    if len(args) > 0 and args[0] == 'game':
+        if args[1] == 'new':
+            if new_game or current_game:
+                client.msg(channel, "Game already created or in progress, the host can end the current game with: !j game end")
+                return
+            else:
+                client.msg(channel, "Fetching questions and categories from the API. This will take ~10s")
+                reactor.callLater(1,setup_new_game, client, channel, nick, message, cmd, args)
+            return
+
+    if len(args) == 2 and args[0] == 'game' and args[1] == 'end':
+        if not new_game and not current_game:
+            client.msg(channel, "No active game, try: !j game new")
+            return
+        if current_game:
+            if nick == current_game['game_host']:
+                client.msg(channel, "Host is ending game...")
+                end_game(client, channel, current_game)
+                return
+            else:
+                client.msg(channel, "Only the game host, {}, can end the game. As a failsafe, this game will end 30 minutes after start.".format(current_game['game_host']))
+        if new_game and not current_game:
+            if nick == new_game['game_host']:
```

Comment:
Should anyone (instead of just the "host") be allowed to end a game if it was created but not started?

Reply:
I'm struggling with the best way to approach this. It seems like that would open the door for some trolling to take place. I think the best way to approach ending the game is a voting system. If anyone says 'game end' we first confirm that they are an active player, then increment a 'vote' column in the db and then divide votes by number of players and end the game if the result is > 0.49. then either player can kill a 2 player match, otherwise it's a majority vote(assuming we don't have 100 players, and we likely won't).       also, the way it is currently, anyone can do a !j reset and it will kill any game whether started or not.